pandas/core/groupby/ops.py

```diff
@@ -835,10 +848,13 @@ def _ob_index_and_ids(
         levels: list[Index],
         codes: list[npt.NDArray[np.intp]],
         names: list[Hashable],
+        sorts: list[bool],
     ) -> tuple[MultiIndex, npt.NDArray[np.intp]]:
+        consistent_sorting = all(sorts[0] == sort for sort in sorts[1:])
```

Comment:
Are we guaranteed for `sorts` to not be empty?

Reply:
Yes - the length is the number of groupings in the groupby, so must be nonzero.